js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetLiveDataProvider.tsx

```diff
@@ -88,35 +88,24 @@ export function useAssetsLiveData(
 }
 
 export const AssetLiveDataProvider = ({children}: {children: React.ReactNode}) => {
-  const [allObservedKeys, setAllObservedKeys] = React.useState<AssetKeyInput[]>([]);
+  const [keysChanged, updateKeysChanged] = useReducer((s) => s + 1, 0);
 
-  const staleKeysObserved = useRef<string[]>([]);
-  const baseKeysObserved = useRef<string[]>([]);
-  const healthKeysObserved = useRef<string[]>([]);
+  const staleKeysObserved = useRef<Set<string>[]>([]);
+  const baseKeysObserved = useRef<Set<string>[]>([]);
+  const healthKeysObserved = useRef<Set<string>[]>([]);
 
   React.useEffect(() => {
-    const onSubscriptionsChanged = () => {
-      const keys = Array.from(
-        new Set([
-          ...staleKeysObserved.current,
-          ...baseKeysObserved.current,
-          ...healthKeysObserved.current,
-        ]),
-      );
-      setAllObservedKeys(keys.map((key) => ({path: key.split('/')})));
-    };
-
     AssetStaleStatusData.manager.setOnSubscriptionsChangedCallback((keys) => {
       staleKeysObserved.current = keys;
-      onSubscriptionsChanged();
+      updateKeysChanged();
     });
     AssetBaseData.manager.setOnSubscriptionsChangedCallback((keys) => {
       baseKeysObserved.current = keys;
-      onSubscriptionsChanged();
+      updateKeysChanged();
     });
     AssetHealthData.manager.setOnSubscriptionsChangedCallback((keys) => {
       healthKeysObserved.current = keys;
-      onSubscriptionsChanged();
+      updateKeysChanged();
     });
   }, []);
```

Comment:
I'm basically holding off on converting the sets into array until the useThrottledEffect call to minimize memory pressure

Reply:
Nice this is smooth 👏